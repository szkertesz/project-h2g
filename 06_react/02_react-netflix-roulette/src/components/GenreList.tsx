import classes from './GenreList.module.scss'

interface Props {
    genres: string[];
}
const GenreList: React.FC<Props> = ({genres}) => {
    return (
        <ul className={classes['genre-list']}>
            {genres.map((genreItem, i, genres) => {
                return (
                    <li key={i}>{`${genreItem}${
                        i < genres.length - 1 ? `, ` : ``
                    }`}</li>
                );
            })}
        </ul>
    );
};

export default GenreList