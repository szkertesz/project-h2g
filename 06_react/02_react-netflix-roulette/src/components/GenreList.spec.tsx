import React from 'react';
import GenreList from './GenreList';
import { render, screen } from '@testing-library/react';

describe('GenreList', () => { // the test suite
    it('should render component', () => { // the test case
        const genres = ['a', 'b', 'c'];
        const { container } = render(<GenreList genres={genres} />);

        // screen.debug();
        expect(container).toMatchSnapshot(); // the assertion
    });
});
