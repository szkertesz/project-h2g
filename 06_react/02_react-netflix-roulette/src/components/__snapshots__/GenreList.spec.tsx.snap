// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList should render component 1`] = `
<div>
  <ul
    class="genre-list"
  >
    <li>
      a, 
    </li>
    <li>
      b, 
    </li>
    <li>
      c
    </li>
  </ul>
</div>
`;
